### ISSUE TITLE
Approved Hidden Comment Displays Incorrectly Until Refresh

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#22024]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Report a comment as Intimidation or Bullying
2. As Concierge, approve the message (i.e. clear the flag)
## Expected Result: 
The message should appear as it did before the flag
## Actual Result: 
The message - until refreshing the page - appears as the "This message has been flagged as violating our community rules and the content has been hidden." copy, without the reveal button or line on the side.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [x] Android / Chrome
- [x] iOS / native
- [x] iOS / Safari
- [x] MacOS / Chrome / Safari
- [x] MacOS / Desktop

**Version Number:** 1.3.35-6
**Reproducible in staging?:**  Needs reproduction requires concierge contribution 
**Reproducible in production?:** Needs reproduction requires concierge contribution 
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/403dabb6-796f-4a64-ba91-286abfe51aad

**Expensify/Expensify Issue URL:**
**Issue reported by:** @dangrous 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1688151033866789

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sonialiap** [comment]
asking follow up question in slack

**Comment by sonialiap** [comment]
Daniel says that he expects the solution to be external but that it will be hard to test without internal access (to Concierge to clear flags) so might need to be a mix of internal and external. Starting off with the internal flag 🎏

**Comment by sonialiap** [comment]
Hmm I think the internal label was supposed to assign an internal engineer. Adding engineering label to assign someone

**Comment by puneetlath** [comment]
@dangrous I'm going OOO for a week and a half starting thursday. Since this bug deals with moderation, I'm going to assign over to you since I believe you implemented that. Please let me know if I'm mistaken though!

**Comment by dangrous** [comment]
yep! my report of a bug with my feature haha. I'll see what I can do.

I'm wondering if this might be a good opportunity to pair with an external engineer, I can let them use my ngrok backend and I can handle the concierge side when it's ready to test. I think that would be pretty easy, but I'm not sure we've officially done that before

**Comment by puneetlath** [comment]
That's cool! I think @MonilBhavsar and @chiragsalian have done that before.

**Comment by dangrous** [comment]
Great, switching this to external

# Hi Contributors!
Because you'll need a moderator to respond to your flag in order to test this, I'm here for you! Once your proposal is accepted, I can provide you with a link you can use to my backend, and I can be your Concierge agent and help test. And, I think this is clear but what happens if a flag is cleared is a new `moderationDecision` is added to the message, with the status of `approved`. That should hopefully be all you need to diagnose and create a proposal, but if you have questions let me know.

**Comment by b4s36t4** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Approved Hidden Comment Displays Incorrectly Until Refresh
### What is the root cause of that problem?

The root cause of the issue lies here in this code block.
https://github.com/Expensify/App/blob/a047b88851917807b036f3c5b8564c5add9ea9d2/src/pages/home/report/ReportActionItem.js#L158-L167

If we observe the code carefully we have a state called `isHidden` in the `useEffect` hook we only setting the value as `true` to `isHidden` state. Means even in case of update `false` we're not overriding the state.

Typically what we're doing is we're only setting state value as `true` we're not resetting the state if the `moderationDecision` changes which means till we refresh the state value going to be `true` and it's staying same.

### What changes do you think we should make in order to solve the problem?

We should add a else check in the `useEffect` hook,  and reset the state value to `false`. There can be many ways to set it to false but the final solution is to make it `false` works.



### What alternative solutions did you explore? (Optional)
NA

## Result
[iu.webm](https://github.com/Expensify/App/assets/59088937/938e56e0-80c1-4877-adc6-56522e4512c4)

**Comment by dangrous** [comment]
@Santhosh-Sellavel let us know when you've had a chance to review the proposals . Seems pretty straightforward, I think!

**Comment by Santhosh-Sellavel** [comment]
@b4s36t4's proposal looks good! @dangrous

**Comment by dangrous** [comment]
Cool! I'm pretty sure just doing it where @b4s36t4 pointed it out should be fine; we'll only ever get a cleared flag from the backend so latestDecision will change whenever we would want to put a comment back to normal.

Given that it's Friday, though, and to test this we need some back and forth and coordination with my backend, I'm not going to assign until Monday (so we should be in a good position for the 3-day bonus rule). But I will do so then!

**Comment by Santhosh-Sellavel** [comment]
@dangrous It's 3 business days (72 business hours), so weekends are excluded from the count. If we can get something to get started. I am sure we can wrap this up Tuesday, what do you say?

**Comment by dangrous** [comment]
oh! For some reason I thought it counted weekends. Sweet, I'll assign now. I think that should be fine. @b4s36t4 ping me when you're ready to test, we can maybe find some time early Monday!

**Comment by b4s36t4** [comment]
@dangrous I think I'm done from my end. Please do let me know when we can connect for backend connection.

**Comment by dangrous** [comment]
I'll be available from 9am-5pm US Central Time (UTC -5) on Monday, if there's a time in there that can work? If that's too tricky I can probably find some time outside of that too. Just let me know! (you can ping me in the app too, if that's easier than going back and forth here)

**Comment by b4s36t4** [comment]
Ok will connect with you on Monday at 10AM(your time) after that.

**Comment by dangrous** [comment]
Sounds great!

**Comment by dangrous** [comment]
Shush Melvin, we're testing and will have PR up soon!

**Comment by dangrous** [comment]
We're having some trouble testing, but it's with my development environment - not anything on @b4s36t4's side, so any delay shouldn't affect the potential speed bonus.

Hopefully I'll be able to get it sorted today and we can still merge tomorrow, but wanted to get ahead of it if not

**Comment by dangrous** [comment]
Still having issues unfortunately, continuing to ask for internal engineer assistance on troubleshooting. This has actually been an issue with my dev environment for a while haha but it just has never mattered before (Pusher does not seem to send Concierge updates).

**Comment by dangrous** [comment]
@b4s36t4 helped me figure out a way around my backend issues, so we've got a slightly janky way to test this, even without fixing my overall error. PR is coming shortly - looks like the code works!

**Comment by dangrous** [comment]
This is on prod! Hopefully we'll get the automated comment soon, but writing here for posterity.

**Comment by sonialiap** [comment]
@b4s36t4 fix+bonus $1500
@Santhosh-Sellavel review+bonus $1500 - please complete the [checklist](https://github.com/Expensify/App/issues/22024#issuecomment-1658917483) and request payment via NewDot

**Comment by Santhosh-Sellavel** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@Santhosh-Sellavel] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/19476
* [x]  [@Santhosh-Sellavel] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/19476/files#r1289489139
* [ ]  [@Santhosh-Sellavel] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: 
* [ ]  [@Santhosh-Sellavel] Determine if we should create a regression test for this bug. 
* [ ]  [@Santhosh-Sellavel] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
* [ ]  [@sonialiap] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by Santhosh-Sellavel** [comment]
Will we need a regression test here?

**Comment by dangrous** [comment]
So I don't think this would have been caught during existing testing, but I think it's next to impossible to have as a regular test due to - as you mentioned, @Santhosh-Sellavel - the fact that we'd need Concierge to respond. I think we can skip it

**Comment by JmillsExpensify** [comment]
Reviewed the details for @Santhosh-Sellavel. Approved for payment in NewDot based on the BZ summary above.
